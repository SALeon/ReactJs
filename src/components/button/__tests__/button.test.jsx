import React from 'react';
import { shallow } from 'enzyme';
import Button from '../index';

const mockProps = {
  clickHandler: jest.fn(),
  label: 'label',
};
describe('Button', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Button
      label={mockProps.label}
      clickHandler={mockProps.clickHandler}
    />);
  });

  it('Button snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('button click', () => {
    wrapper.find('.button').simulate('click');
    expect(mockProps.clickHandler).toHaveBeenCalled();
  });
});
