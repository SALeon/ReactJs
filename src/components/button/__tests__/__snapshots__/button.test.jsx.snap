// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button snapshot 1`] = `
<button
  className="button "
  onClick={[MockFunction]}
  type="submit"
>
  label
</button>
`;
